src/switch.js

```diff
@@ -0,0 +1,271 @@
+     * @private
+     */
+    this.multiSelectGroup_ = null;
+
+    /**
+     * Whether this has been initialized.
+     * @type {boolean}
+     * @private
+     */
+    this.initialized_ = false;
+  }
+  /**
+   * Create the multi select controls.
+   * @return {!SVGElement} The multi select controls SVG group.
+   */
+  createDom() {
+    this.svgGroup_ = Blockly.utils.dom.createSvgElement(
+        Blockly.utils.Svg.G, {}, null);
+    this.createMultiSelectSvg_();
+    return this.svgGroup_;
+  }
+  /**
+   * Initializes the multi select controls.
+   */
+  init() {
+    this.workspace_.getComponentManager().addComponent({
+      component: this,
+      weight: 3,
+      capabilities: [Blockly.ComponentManager.Capability.POSITIONABLE],
+    });
+    this.initialized_ = true;
+    this.workspace_.resize();
+  }
+  /**
+   * Disposes of this multi select controls.
+   * Unlink from all DOM elements to prevent memory leaks.
+   */
+  dispose() {
+    this.workspace_.getComponentManager().removeComponent(this.id);
+    if (this.svgGroup_) {
+      Blockly.utils.dom.removeNode(this.svgGroup_);
+    }
+    if (this.onMultiSelectWrapper_) {
+      Blockly.browserEvents.unbind(this.onMultiSelectWrapper_);
+    }
+  }
+  /**
+   * Returns the bounding rectangle of the UI element in pixel units relative to
+   * the Blockly injection div.
+   * @return {?Blockly.utils.Rect} The UI elements's bounding box. Null if
+   *   bounding box should be ignored by other UI elements.
+   */
+  getBoundingRectangle() {
+    const bottom = this.top_ + HEIGHT;
+    const right = this.left_ + WIDTH;
+    return new Blockly.utils.Rect(this.top_, bottom, this.left_, right);
+  }
+  /**
+   * Positions the multi select controls.
+   * It is positioned in the opposite corner to the corner the
+   * categories/toolbox starts at.
+   * @param {!Blockly.MetricsManager.UiMetrics} metrics The workspace metrics.
+   * @param {!Array<!Blockly.utils.Rect>} savedPositions List of rectangles that
+   *     are already on the workspace.
+   */
+  position(metrics, savedPositions) {
+    // Not yet initialized.
+    if (!this.initialized_) {
+      return;
+    }
+
+    const cornerPosition =
+      Blockly.uiPosition.getCornerOppositeToolbox(this.workspace_, metrics);
+    const startRect = Blockly.uiPosition.getStartPositionRect(
+        cornerPosition, new Blockly.utils.Size(WIDTH, HEIGHT),
+        MARGIN_HORIZONTAL,
+        MARGIN_VERTICAL, metrics, this.workspace_);
+
+    const verticalPosition = cornerPosition.vertical;
+    const bumpDirection = verticalPosition ===
+      Blockly.uiPosition.verticalPosition.TOP ?
+      Blockly.uiPosition.bumpDirection.DOWN :
+      Blockly.uiPosition.bumpDirection.UP;
+    const positionRect = Blockly.uiPosition.bumpPositionRect(
+        startRect, MARGIN_VERTICAL, bumpDirection, savedPositions);
+
+    if (verticalPosition === Blockly.uiPosition.verticalPosition.TOP) {
+      this.MultiSelectGroup_.setAttribute(
```

Comment:
Why are you positioning this differently when the position is `top`? Have you tested this will all workspace configurations? (vertical, horizontal, start, end, rtl, and ltr)

Reply:
Yes, I've tested it with those workspace configurations:

![](https://user-images.githubusercontent.com/43995067/179136432-0ac60f70-9877-452c-9ccf-c35a7917042b.png)
![](https://user-images.githubusercontent.com/43995067/179136501-e634d775-8925-4472-bf3f-331bb655a1e9.png)
![](https://user-images.githubusercontent.com/43995067/179136515-cd3196f1-c0ae-49bf-a2d0-a10fdadadc82.png)
![](https://user-images.githubusercontent.com/43995067/179136521-5a44ebbb-450a-44ed-abb7-75925517c7f4.png)

Reply:
Nice job testing with the different configurations!

I'm still confused why you move the button down further when it's positioned at the top though.

Reply:
Oh, okay, I also don't remember and can't find a reason for this 😂, so will remove this then.